### DISCUSSION TITLE
Stuck with a main.lua filter error  rendering a Quarto complex project

### DISCUSSION THREAD

**Discussion by r-scotti** [quarto-dev/quarto-cli#9546] [Q&A] [answered]
### Description

Hello,
   I am rendering a document that has 'childs' and 'childs' of 'childs' and I am now stuck with the following lua error.   The document used to render before I updated parts of my environment.   
I do not how to reduce the code to a simple "Reproducible Quarto Document" still throwing the error.

When I click on the Render button the process flows well up to the following step

>
>        .... (omitted) ..............
> output file: 2018MarganaiSoilErosionExperimentalEvaluation_Copernicus.knit.md
> **pandoc** 
>   to: latex
>   output-file: 2018MarganaiSoilErosionExperimentalEvaluation_Copernicus.tex
>   standalone: true
>   pdf-engine: pdflatex
>   variables:
>     graphics: true
>     tables: true
>   default-image-extension: pdf  
>
>        .... (omitted) ..............
>   citation: true
>   bibliography:
>     - bib/references.bib
>   
> **Error running filter /opt/quarto/share/filters/main.lua:**
> /opt/quarto/share/filters/main.lua:15155: **attempt to get length of a boolean value (upvalue 'subcaps')**
> stack traceback:
> 	[C]: in ?
> 	[C]: in method 'walk'
> 	/opt/quarto/share/filters/main.lua:224: in function </opt/quarto/share/filters/main.lua:214>
> 	(...tail calls...)
> 	/opt/quarto/share/filters/main.lua:15153: in local 'fn'
> 	/opt/quarto/share/filters/main.lua:2554: in function </opt/quarto/share/filters/main.lua:2548>
> 	(...tail calls...)
> ...... (omitted last lines are repeated)

My environment:
*  quarto --version  -->  1.4.553
* OS   Linux Ubuntu 20.04 

If it is not possible to identify and solve the problem with just the above information, please instruct on how I can provide more.  
Many thanks for any help

**Comment by mcanouil** [comment]
From the log "subcaps", you should check for figures/tables sub captions.

Without any code, we can't do more.

**Comment by r-scotti** [comment]
I have figures and tables but no subcaptions.
I understand the requirement for code, in case I can share the project, it
is on GitLab, but I am afraid it is not the right way.
Can you suggest how to focus on the relevant lines?
Many thanks


Roberto Scotti
Pubblicazioni:
https://www.zotero.org/groups/2543874/prof._roberto_scotti

Il Mer 1 Mag 2024, 13:56 Mickaël Canouil ***@***.***> ha
scritto:

> From the log "subcaps", you should check for figures/tables sub captions.
>
> Without any code, we can't do more.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/quarto-dev/quarto-cli/discussions/9546#discussioncomment-9284526>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEAKXD74NPS5PZ7UMLIHG7DZADKAHAVCNFSM6AAAAABHB3LBIGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TEOBUGUZDM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 

--
*Dona il  5x1000* all'Università degli Studi di Sassaricodice fiscale: 
00196350904

**Reply by mcanouil** [reply]
If you don't share code there is nothing we can do.

You need to debug this yourself.
The usual method is either to start small and add small pieces or remove small pieces and render until you find where is the issue.
We unfortunately don't have the time to do this process for you.

**Comment by r-scotti** [comment, marked as answer]
I made some progress.
Inserting the following chunk 
> ```{r}
> save.image("phase2.Rdata")
> knit_exit()
> 
> ```
I found a way to create a kind of breakpoint. 
Moving it through the Quarto file I could spot the chunk(s) that would abort the rendering.
Apparently (in contrast with the example in https://quarto.org/docs/authoring/cross-references.html ) the 
> #| tbl-subcap: true   

option was causing the problem! 
If I remove `true` and specify the sub captions adding
>        - a
>        - b
the rendering completes.
```
---
title: "Reproducible Quarto Document"
format: pdf
editor: visual
---

```{r}
#| label: tbl-tables
#| tbl-cap: "Tables"
#| tbl-subcap:
#|  - a
#|  - b
#| layout-ncol: 2

library(knitr)
kable(head(cars))
kable(head(pressure))
```

```

**Reply by mcanouil** [reply]
I need to check because it should work, if not (even on the development version), then there is a bug (or even a regression) in Quarto.

**Reply by cscheid** [reply]
I can reproduce a crash. To be clear, `fig-subcap: true` doesn't make sense in your context, because the figures you're producing do not have captions. `fig-subcap: true` exists to tell Quarto that the captions you've provided are meant to be interpreted as subcaptions. If you haven't provided captions, then there's nothing that Quarto can do.

We shouldn't crash, but you also need to fix your code.

**Reply by mcanouil** [reply]
I am creating an issue for this.

Note that it's documented at https://quarto.org/docs/authoring/cross-references.html#floats
<img width="600" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/c429fff0-8fcc-411e-97e7-7dab5c066032">
<img width="600" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/f8843009-73b9-4886-a068-3d237f81e0a5">

**Reply by mcanouil** [reply]
- #9550

**Reply by cscheid** [reply]
Oh, huh. That's obnoxious. My understanding of the intent of `fig-subcap: true` is so that if your embedded figures (or tables) have captions in them, we know to interpret them as subcaptions rather than multiple full figures with captions.

Now I need to confirm this with the other devs...
